src/components/Settings/index.jsx

```diff
@@ -114,7 +118,26 @@ export class Settings extends Component {
         shouldDisplayViewOfferButton =
           instanceModel.shouldDisplayOffers(data) || hasAnOffer(data)
 
-        managerUrlPremiumLink = instanceModel.buildPremiumLink(data)
+        const hasSubscription = flag('settings.subscription')
+
+        if (hasSubscription) {
+          // Contrived but we don't have access to client hooks nor client itself
+          stack.get.capabilities().then(res => {
+            const webLink = generateWebLink({
+              cozyUrl: stack.get.cozyURL(),
+              subDomainType: res.data.attributes.flat_subdomains
```

Comment:
I think you don't need to get `capabilities` with a request there already into the cozy-client instance you can access them with :
```
const { subdomain: subDomainType } = client.getInstanceOptions()
```
We're trying to do without redux in the cozy-bar as much as possible. To avoid feeding the monster